build/xtask/src/dist.rs

```diff
@@ -549,6 +555,70 @@ pub fn package(
     Ok(allocated)
 }
 
+// can we use lpc55-romapi
+const LPC55_FLASH_PAGE_SIZE: usize = 512;
+
+// generate file with hash of expected flash contents
+fn write_fwid(
+    cfg: &PackageConfig,
+    image_name: &String,
+    flash: &Range<u32>,
+    archive_name: &PathBuf,
+) -> Result<()> {
+    let mut archive = hubtools::RawHubrisArchive::load(archive_name)
+        .context("loading archive with hubtools")?;
+
+    let bin = archive
+        .extract_file("img/final.bin")
+        .context("extracting final.bin after signing & caboosing")?;
+
+    let chip_name = Path::new(&cfg.toml.chip);
+
+    // determine length of padding
+    let pad = match chip_name.file_name().and_then(OsStr::to_str) {
+        Some("lpc55") => {
+            // Flash is programmed in 512 blocks. If the final block is not
+            // filled, it is padded with 0xff's. Unwritten flash pages cannot
+            // be read and are not included in the FWID calculation.
+            LPC55_FLASH_PAGE_SIZE - bin.len() % LPC55_FLASH_PAGE_SIZE
+        }
+        Some("stm32h7") => {
+            // all unprogrammed flash is read as 0xff
+            flash.end as usize - flash.start as usize - bin.len()
+        }
+        Some(c) => {
+            bail!("no FWID algorithm defined for chip: \"{}\"", c)
+        }
+        None => bail!("Failed to get file name of {}", chip_name.display()),
+    };
+
+    let mut sha = Sha3_256::new();
+    sha.update(&bin);
+
+    if pad != 0 {
+        sha.update(vec![0xff_u8; pad])
+    }
+
+    let digest = sha.finalize();
+
+    // after we've appended a newline fwid is immutable
+    let mut fwid = hex::encode(&digest);
+    writeln!(fwid, "").context("appending newline to FWID")?;
+    let fwid = fwid;
```

Comment:
Is the newline just for ease of reading or is there some deeper meaning?

Reply:
No deeper meaning, just gives better output when the file is dumped to the console.